src/troubleshooting/content.md

```diff
@@ -0,0 +1,641 @@
+**TL;DR:** here's a diagram to help you debug your deployments in Kubernetes.
+
+![Flow chart to debug deployments in Kubernetes](diagram.png)
+
+When you wish to deploy an application in Kubernetes, you usually define three components:
+
+- a Deployment — which is a recipe for creating copies of your application called Pods
+- a Service — an internal load balancer that routes the traffic to Pods
+- an Ingress — a description of how the traffic should flow from outside the cluster to your Service.
+
+Here's a quick visual recap.
+
+```slideshow
+{
+  "description": "Kubernetes core concepts: Ingress, Service, Pod and Deployment.",
+  "slides": [
+    {
+      "image": "core-concepts-1.svg",
+      "description": "In Kubernetes your applications are exposed through two layers of load balancers: internal and external."
+    },
+    {
+      "image": "core-concepts-2.svg",
+      "description": "The internal load balancer is called Service, whereas the external one is called Ingress."
+    },
+    {
+      "image": "core-concepts-3.svg",
+      "description": "Pods are not deployed directly. Instead, the Deployment creates the Pods and whatches over them."
+    }
+  ]
+}
+```
```

Comment:
I would delete slide 1 and 2.  In courses it's very good to have a lot of slides, but blogs are read on small iphones and android phones and not everyone will click next two times to get the correct picture.  I also hate the first slide that it contains the "External Load Balancer == Ingress" notion, as that is "external router" and calling something load balancer that is a router causes always confusion at clients.

Reply:
I'm happy to change the name to "router". I think it's important to present concepts one at the time.

Reply:
It's true that an Ingress has more the function of a router than of a load balancer. It forwards request to the correct destination, rather than balancing them across a set of equivalent destinations.